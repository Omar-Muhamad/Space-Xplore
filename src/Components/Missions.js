import React from 'react';
import { useSelector } from 'react-redux';
import styles from './Missions.module.css';
import MissionItem from './MissionItem';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  return (
    <table
      className={`${styles.table} border border-slate-300 sm:mx-12 mx-0 my-2 mt-12`}
    >
      <thead>
        <tr>
          <th className="border border-slate-300">Mission</th>
          <th className="border border-slate-300">Description</th>
          <th className="border border-slate-300">Status</th>
          <th className="border border-slate-300"> </th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <MissionItem key={mission.mission_id} mission={mission} />
        ))}
      </tbody>
    </table>
  );
};

export default Missions;
