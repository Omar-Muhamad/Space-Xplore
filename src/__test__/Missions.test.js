import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../Redux/configreStore';
import Missions from '../Components/Missions';

describe('Test Missions render', () => {
  test('Test Missions', () => {
    const tree = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
