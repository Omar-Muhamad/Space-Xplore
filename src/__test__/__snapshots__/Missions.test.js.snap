// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions render Test Missions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table border border-slate-300 md:mx-12 mx-0 my-2 mt-12"
      >
        <thead>
          <tr>
            <th
              class="border border-slate-300"
            >
              Mission
            </th>
            <th
              class="border border-slate-300"
            >
              Description
            </th>
            <th
              class="border border-slate-300"
            >
              Status
            </th>
            <th
              class="border border-slate-300"
            >
               
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table border border-slate-300 md:mx-12 mx-0 my-2 mt-12"
    >
      <thead>
        <tr>
          <th
            class="border border-slate-300"
          >
            Mission
          </th>
          <th
            class="border border-slate-300"
          >
            Description
          </th>
          <th
            class="border border-slate-300"
          >
            Status
          </th>
          <th
            class="border border-slate-300"
          >
             
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
